src/services/resource.js

```diff
@@ -210,6 +327,394 @@ module.exports = class resourceHelper {
+				delete item.organization_id
+				delete returnValue.meta
+				return returnValue
+			})
+			result.count = response.count
+			result.in_progress_count = in_progress_count
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'RESOURCE_LISTED_SUCCESSFULLY',
+				result,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async findSequentialResources(organization_id, roles, resourceTypes = []) {
+		// get unique user roles
+		const userRoleTitles = utils.getUniqueElements(roles.map((item) => item.title))
+
+		// fetch the resource wise review levels
+		const resourceWiseLevels = await this.fetchResourceReviewLevel(organization_id, userRoleTitles, resourceTypes)
+		let resourceTypeStagesConfig = []
+		resourceTypes.filter((type) => {
+			if (resourceWiseLevels[type]) {
+				resourceTypeStagesConfig.push({
+					[Op.and]: [{ type: type }, { next_stage: resourceWiseLevels[type] }],
+				})
+			}
+		})
+		const resourceFilter = {
+			[Op.or]: resourceTypeStagesConfig,
+			organization_id,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.findAll(resourceFilter, ['id'])
+		if (resourcesDetails) {
+			const resoureId = resourcesDetails.map((item) => {
+				return item.id
+			})
+			return resoureId
+		}
+		return []
+	}
+
+	static async findParallelResources(organization_id, resourceTypes = []) {
+		const resourceFilter = {
+			type: {
+				[Op.in]: resourceTypes,
+			},
+			organization_id,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.findAll(resourceFilter, ['id'])
+		if (resourcesDetails) {
+			const resoureId = resourcesDetails.map((item) => {
+				return item.id
+			})
+			return resoureId
+		}
+		return []
+	}
+	// fetch all the resource ids from reviews table which is assigned to the reviewer
+	//  and already started reviewer
+	static async findResourcesAssignedToReviewer(organization_ids, logged_in_user_id) {
+		const reviewsFilter = {
+			organization_id: { [Op.in]: organization_ids },
+			reviewer_id: logged_in_user_id,
+			status: { [Op.in]: common.PAGE_STATUS_VALUES.up_for_review },
+		}
+
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+		if (reviewsResponse.length > 0) {
+			const res = reviewsResponse.map((item) => {
+				return item.resource_id
+			})
+			return res
+		} else {
+			return []
+		}
+	}
+
+	static async findResourcesPickedUpByAnotherReviewer(loggedInUserId, openToAllResourcesMatchingMyLevel) {
+		// remove all the resouces in sequential review picked up by another reviewer
+		const reviewsFilter = {
+			resource_id: { [Op.in]: openToAllResourcesMatchingMyLevel },
+			status: {
+				[Op.in]: [
+					common.REVIEW_STATUS_INPROGRESS,
+					common.REVIEW_STATUS_CHANGES_UPDATED,
+					common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
```

Comment:
@adithyadinesh0412 if the reviewer is requested for changed then that resource no need to show in the listing

Reply:
yes, we are removing the ids in these statuses. This is within the findResourcesPickedUpByAnotherReviewer() which will be used to find out what all resources we should omit out from the users list. So resources which are picked up by other users in these statuses will be removed from the final list @priyanka-TL